### PR TITLE
Add @Inject annotation to http config classes

### PR DESCRIPTION
**Pull request by wesleyk**

### CODE REVIEW

**Comment by wesleyk** [comment]
Needed to be able to @Provide some default configs as part of building dynamic clients

File: misk-core/src/main/kotlin/misk/security/ssl/CertStoreConfig.kt [hunk 1]
```diff
@@ -1,6 +1,8 @@
 package misk.security.ssl
 
-data class CertStoreConfig(
+import javax.inject.Inject
+
+data class CertStoreConfig @Inject constructor(
   val resource: String,
```
Comment (zcross) [inline misk-core/src/main/kotlin/misk/security/ssl/CertStoreConfig.kt]: How are these strings w/out annotations selected for injection?

File: misk-core/src/main/kotlin/misk/security/ssl/TrustStoreConfig.kt [hunk 2]
```diff
@@ -1,8 +1,9 @@
 package misk.security.ssl
 
 import misk.security.ssl.SslLoader.Companion.FORMAT_JCEKS
+import javax.inject.Inject
 
-data class TrustStoreConfig(
+data class TrustStoreConfig @Inject constructor(
```
Comment (zcross) [inline misk-core/src/main/kotlin/misk/security/ssl/TrustStoreConfig.kt]: Same Q: are these all provided by factories or providers that invoke the constructor manually?

File: misk/src/main/kotlin/misk/client/HttpClientsConfig.kt [hunk 3]
```diff
@@ -50,7 +51,7 @@ data class HttpClientEndpointConfig(
   val ssl: HttpClientSSLConfig? = null
 )
 
-data class HttpClientEnvoyConfig(
+data class HttpClientEnvoyConfig @Inject constructor(
   val app: String,
```
Comment (zcross) [inline misk/src/main/kotlin/misk/client/HttpClientsConfig.kt]: Does this need `@AppName`?

File: misk-core/src/main/kotlin/misk/security/ssl/TrustStoreConfig.kt (hunk 2 above)
Comment (wesleyk) [reply misk-core/src/main/kotlin/misk/security/ssl/TrustStoreConfig.kt]: These aren't injected out of the box. Explicit instances are provided by the app from HttpClientsConfig. However, our strict guice mode means that you need to have this annotation, even if you provide it.

Comment (zcross) [reply misk-core/src/main/kotlin/misk/security/ssl/TrustStoreConfig.kt]: Ah, that clarifies things. Thanks!

File: misk/src/main/kotlin/misk/client/HttpClientsConfig.kt [hunk 4]
```diff
@@ -29,14 +30,30 @@ data class HttpClientsConfig(
         ssl = endpointConfig.ssl ?: ssl
     )
   }
+
+  /**
+   * Returns a default config with no url/envoy config specified.
+   * Used to build up endpoint config for clients dynamically.
+   */
+  fun getDefault(): HttpClientEndpointConfig {
```
Comment (zcross) [inline misk/src/main/kotlin/misk/client/HttpClientsConfig.kt]: Ah, this is nice.

File: misk-core/src/main/kotlin/misk/security/ssl/CertStoreConfig.kt [hunk 5]
```diff
@@ -1,6 +1,8 @@
 package misk.security.ssl
 
-data class CertStoreConfig(
+import javax.inject.Inject
+
+data class CertStoreConfig @Inject constructor(
```
Comment (swankjesse) [inline misk-core/src/main/kotlin/misk/security/ssl/CertStoreConfig.kt]: I don’t think this is going to inject the strings you want it to inject

File: misk-core/src/main/kotlin/misk/security/ssl/TrustStoreConfig.kt (hunk 2 above)
Comment (swankjesse) [reply misk-core/src/main/kotlin/misk/security/ssl/TrustStoreConfig.kt]: Oh wow, that’s a terrible bug. Lets fix

**Pull request merged**